### ISSUE TITLE
网络结构中SpatialFeatureEnhance有许多版本，且与论文都不同

### ISSUE THREAD

**Issue by zzzzzzzcccc** [LJXLJXLJXLJ/Crossmodal-Feature-Fusion-via-Mutual-Assistance-A-Novel-Network-for-Enhanced-Object-Detection#2]
作者您好！感谢您所做的工作！
在models/transformer/net.yaml中 
1.使用了SimAM但是common.py中有许多版本的SimAM，如SimAM，SimAM_conv_256，SimAM_conv_512，加卷积和不加卷积的区别。
2.在net.yaml中不同尺度使用了SpatialFeatureEnhance_V1_sftv1，SpatialFeatureEnhance_V1_512_sftv1，SpatialFeatureEnhance_V1_1024_sftv1，有何不同，而且在common.py中未找到SpatialFeatureEnhance_V1_512_sftv1，SpatialFeatureEnhance_V1_1024_sftv1，在common.py中存在许多版本的SpatialFeatureEnhance，作用有何不同，似乎每一个SpatialFeatureEnhance的网络结构都与论文中的网络结构不符。

如果可以，我希望您提供正确的net.yaml和model.py。

以下是提供的net.yaml的backbone部分
backbone:
  # [from, number, module, args]
  # Two Stream
  [
    ######### TransformerBlock One #############
    # stream one
    [-1, 1, Focus, [64, 3]],    # 0-P1/2
    [-1, 1, Conv, [128, 3, 2]], # 1-P2/4
    [-1, 3, C3, [128]],   # 2-P2/4
    [-1, 1, Conv, [256, 3, 2]],  # 3-P3/8
    [-1, 9, C3, [256]],   # 4-P3/8
    [-1, 1, SimAM, [256]], #5
    # stream two
    [-4, 1, Focus, [64, 3]],    # 5-P1/2 6
    [-1, 1, Conv, [128, 3, 2]], # 6-P2/4 7
    [-1, 3, C3, [128]],   # 7-P2/4 8
    [-1, 1, Conv, [256, 3, 2]],  # 8-P3/8 9
    [-1, 9, C3, [256]],   # 9-P3/8 10
    [-1, 1, SimAM, [256]], #11

    ######### TransformerBlock Two #############
   # transformer fusion
    [[5,11], 1, SpatialFeatureEnhance_V1_sftv1, [256]],   # 10-P3/8  12
    [[4,12], 1, Add2, [256,0]],    # 11-P3/8 stream one x+trans[0] 13

    [[10,12], 1, Add2, [256,1]],    # 12-P3/8 stream two x+trans[1] 14

   ######### TransformerBlock Three #############
   # stream one
    [13, 1, Conv, [512, 3, 2]],   # 13-P4/16 15
    [-1, 9, C3, [512]],   # 14-P4/16 16
    [-1, 1, SimAM, [512]], #17
   # stream two
    [14, 1, Conv, [512, 3, 2]],   # 15-P4/16 18
    [-1, 9, C3, [512]],   # 16-P4/16 19
    [-1, 1, SimAM, [512]], #20
   # transformer fusion
    [[17,20], 1, SpatialFeatureEnhance_V1_512_sftv1, [512]],   # 17-P3/8 21
    [[16,21], 1, Add2, [512,0]],    # 18-P3/8 stream one x+trans[0] 22
    [[19,21], 1, Add2, [512,1]],    # 19-P3/8 stream two x+trans[1] 23

    ######### TransformerBlock Four #############
    # stream one
    [22, 1, Conv, [1024, 3, 2]],    # 20-P5/32 24
    [-1, 1, SPP, [1024, [5, 9, 13]]],   # 21-P5/32 25
    [-1, 3, C3, [1024, False]],   # 22-P5/32 26
    [-1, 1, SimAM, [512]],   #27
    # stream two
    [23, 1, Conv, [1024, 3, 2]],    # 23-P5/32 28
    [-1, 1, SPP, [1024, [5, 9, 13]]],   # 24-P5/32 29
    [-1, 3, C3, [1024, False]],   # 25-P5/32 30
    [-1, 1, SimAM, [512]], #31
    # transformer fusion
    [[27,31], 1, SpatialFeatureEnhance_V1_1024_sftv1, [1024]],    # 26-P5/32 32
    [[26,32], 1, Add2, [1024,0]],    # 27-P5/32 stream one x+trans[0] 33
    [[30,32], 1, Add2, [1024,1]],    # 28-P5/32 stream two x+trans[1] 34

    ######### Add Block #############
    [[13,14], 1, Add, [1]],   # 29-P3/8 fusion backbone P3 33
    [[22,23], 1, Add, [1]],   # 30-P4/16 fusion backbone P4 34
    [[33,34], 1, Add, [1]],   # 31-P5/32 fusion backbone P5 35

  ]

**Comment by jacksonwu09** [comment]
你好，你跑起来了吗 你知道M3FD数据集是按什么比例划分的train、val、test

**Comment by zzzzzzzcccc** [comment]
> 你好，你跑起来了吗 你知道M3FD数据集是按什么比例划分的train、val、test
大概率是跑不起来的，作者提供的代码似乎存在问题，而且网络结构和论文中的不一致，我按照作者提供的代码根据自己的理解进行了改正，但效果似乎并不好，可能是提供的代码网络结构错误的原因或者其他原因

**Comment by zzzzzzzcccc** [comment]
代码中SpatialFeatureEnhance_V1_sftv1似乎只有对红外光特征的操作是正确的，而可见光特征的操作似乎存在很大的错误，而且代码中对特征图操作的维度存在错误，整体网络结构似乎也与论文中不符，以上只是我个人的理解，希望作者指出错误

**Comment by jacksonwu09** [comment]
> SpatialFeatureEnhance_V1_sftv1

我目前在LLVIP数据集上跑完了，我只是将作者原本的SpatialFeatureEnhance_V1_sftv1类似的模块名去掉_sftv1后缀，然后根据models中有的模块在yaml中进行修改，然后可以跑通，但是最终结果只有96.几，应该是论文作者还没有更新最新的模块代码

**Comment by jacksonwu09** [comment]
还有对于M3FD数据集的划分，我在作者论文中指出的参考14论文中没有找到对于测试集的划分，只提供了train、val的数据集

**Comment by zzzzzzzcccc** [comment]
> > SpatialFeatureEnhance_V1_sftv1
> 
> 我目前在LLVIP数据集上跑完了，我只是将作者原本的SpatialFeatureEnhance_V1_sftv1类似的模块名去掉_sftv1后缀，然后根据models中有的模块在yaml中进行修改，然后可以跑通，但是最终结果只有96.几，应该是论文作者还没有更新最新的模块代码

只是用LLVIP中的红外图像在yolov5上的map50都有96

**Comment by jacksonwu09** [comment]
> > > SpatialFeatureEnhance_V1_sftv1
> > 
> > 
> > 我目前在LLVIP数据集上跑完了，我只是将作者原本的SpatialFeatureEnhance_V1_sftv1类似的模块名去掉_sftv1后缀，然后根据models中有的模块在yaml中进行修改，然后可以跑通，但是最终结果只有96.几，应该是论文作者还没有更新最新的模块代码
> 
> 只是用LLVIP中的红外图像在yolov5上的map50都有96

是的  作者应该还没有将核心代码上传吧，确实很多模块都缺失了，在训练代码中也存在问题，对于rgb_ir的训练代码也是缺失的，我是将原本的训练代码添加上去才跑通的，看看作者后续会不会更新下最新的代码
